src/ORM/DataObjectSchema.php

```diff
@@ -457,6 +485,70 @@ protected function cacheDatabaseFields($class)
     }
 
     /**
+     * Cache all indexes for the given class.
+     * Will do nothing if already cached
+     *
+     * @param $class
+     */
+    protected function cacheDatabaseIndexes($class)
+    {
+        if (array_key_exists($class, $this->databaseIndexes)) {
+            return;
+        }
+        $indexes = [];
+
+        // look for indexable field types
+        foreach ($this->databaseFields($class, false) as $field => $type) {
+            if ($type === 'ForeignKey' || $type === 'DBClassName') {
+                $indexes[$field] = [
```

Comment:
Note for future enhancement: Polymorphic has_ones need indexes on FieldID and FieldClass.